backend/src/main/java/app/linguistai/bmvp/controller/AccountController.java

```diff
@@ -98,4 +105,63 @@ public ResponseEntity<Object> getUsers() {
             return Response.create(ExceptionLogger.log(e), HttpStatus.OK);  
         }        
     }
+
+    @CrossOrigin(origins = "http://localhost:5173", allowedHeaders = "*", allowCredentials = "true")
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, path = "/request-reset")
+    public ResponseEntity<Object> requestResetPassword(@Valid @RequestBody QResetPassword resetPasswordInfo) {
+        try {
+            ResetToken resetToken = accountService.generateEmailToken(resetPasswordInfo.getEmail());
+            emailService.sendPasswordResetEmail(resetPasswordInfo.getEmail(), resetToken);
+            return Response.create("Password reset email is sent", HttpStatus.OK);
+        } catch (Exception e){
+            return Response.create(ExceptionLogger.log(e), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @CrossOrigin(origins = "http://localhost:5173", allowedHeaders = "*", allowCredentials = "true")
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, path = "/test-reset")
+    public ResponseEntity<Object> requestResetPasswordWithoutEmail(@Valid @RequestBody QResetPassword resetPasswordInfo) {
+        try {
+            ResetToken resetToken = accountService.generateEmailToken(resetPasswordInfo.getEmail());
+            return Response.create("Reset token is generated", HttpStatus.OK, resetToken);
+        } catch (Exception e){
+            return Response.create(ExceptionLogger.log(e), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @CrossOrigin(origins = "http://localhost:5173", allowedHeaders = "*", allowCredentials = "true")
+    @GetMapping("/validate-reset")
+    public ResponseEntity<Object> validateResetPassword(@Valid @RequestBody QResetPasswordVerification verificationInfo) {
+        try {
+            boolean tokenValid = accountService.validateResetCode(verificationInfo.getEmail(), verificationInfo.getResetCode(), false);
+            if (tokenValid) {
+                return Response.create("Valid password reset token", HttpStatus.OK);
+            } else {
+                return Response.create("Invalid password reset token", HttpStatus.BAD_REQUEST);
+            }
+        } catch (Exception e){
+            return Response.create(ExceptionLogger.log(e), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @CrossOrigin(origins = "http://localhost:5173", allowedHeaders = "*", allowCredentials = "true")
+    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE, path = "/reset-password")
+    public ResponseEntity<Object> saveResetPassword(@Valid @RequestBody QResetPasswordSave passwordInfo) {
+        try {
+            boolean tokenValid = accountService.validateResetCode(passwordInfo.getEmail(), passwordInfo.getResetCode(), true);
+            if (tokenValid) {
```

Comment:
Instead of nesting if statements, can we use guard clauses here? It will improve code maintainability and readability.

![image](https://github.com/LinguistAI/Linguist-AI-MVP/assets/86242974/2bad74a4-2638-4230-8f52-ed63d0e3175d)

Reply:
Thanks, changed! Looks better

Reply:
Thanks, changed it. Looks much better